File: 6KYU/Bouncing Balls/solution.clj
(ns bouncing-balls.core)

(defn bouncing-balls
  ([h bounce window]
   (if (or (neg? h) (zero? h) (>= window h) (not (< 0 bounce 1)))
     -1
     (bouncing-balls h bounce window 0)))
  ([h bounce window bounces]
   (if (<= h window)
     (inc (* 2 (dec bounces)))
     (recur (* bounce h) bounce window (inc bounces)))))

File: 6KYU/Bouncing Balls/solution.go
package kata

func BouncingBall(h, bounce, window float64) int {
	if h <= 0 || window > h || bounce <= 0 || bounce >= 1 {
		return -1
	}
	bounces := 0
	for {
		if h <= window {
			break
		}

		h = h * bounce
		bounces += 1
	}

	return ((bounces - 1) * 2) + 1
}
